src/hooks/useGetOriginUrl.ts

```diff
@@ -0,0 +1,13 @@
+import { useEffect, useState } from 'react';
+
+export const useGetOriginUrl = (): string | null => {
+  const [origin, setOrigin] = useState<string | null>(null);
+
+  useEffect(() => {
+    if (typeof window !== 'undefined') {
+      setOrigin(window.location.origin);
+    }
+  }, []);
+
+  return origin;
+};
```

Comment:
This hook is almost certainly unnecessary, as it can be replaced in its entirety with:
```js
typeof location === 'undefined' ? null : location.origin
```